src/avro_util.erl

```diff
@@ -62,6 +63,72 @@
 -type validate_fun() :: fun((type_or_name(), validate_hist()) -> ok).
 
 %%%_* APIs =====================================================================
+%% @doc Checks whether two schemas are compatible in sense that an event
+%% encoded with the writer schema can be decoded by the reader schema.
+%% Schema compatibility is defined in
+%% https://avro.apache.org/docs/1.8.1/spec.html#Schema+Resolution
+%% @end
+-spec avro_schema_compatible(avro_type(), avro_type()) ->
+                                true | false.
```

Comment:
instead of returning a boolean().
it's perhaps better to implement this internal function with a `erlang:throw(Reason)`
so the users have better clue which part is not compatible.

then wrap it around a `try ... catch ...` for `avro:is_compatible/2`.